.storybook/util/withState.jsx

```diff
@@ -0,0 +1,19 @@
+// @flow
+
+import React, {type Node, Component} from 'react';
+
+class WithState extends Component<{|initialState: {}, children: Function, refs?: any[]|}, {}> {
+  state = {
+    ...this.props.initialState
+  }
+  componentDidMount() {
+    this.refs = this.props.refs;
+  }
+  render() {
+    return (
+      this.props.children(this.state, (state) => this.setState(state), this.refs)
+    );
+  }
+}
+
+export default WithState;
```

Comment:
I've locked this away in the storybook folder because I don't think live code should have these powers but maybe we _do_ want it? it's a bit like having hooks